src/Microsoft.OData.Core/UrlValidation/ODataUrlValidationMessage.cs

```diff
@@ -0,0 +1,88 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUrlValidationMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.OData.UriParser.Validation
+{
+    /// <summary>
+    /// Severity of an OData Url Validation Message
+    /// </summary>
+    public enum Severity
+    {
+        /// <summary>
+        /// Default / Undefined error Level
+        /// </summary>
+        Undefined = 0,
+
+        /// <summary>
+        /// Informational Message
+        /// </summary>
+        Message = 1,
+
+        /// <summary>
+        /// Warning
+        /// </summary>
+        Warning = 2,
+
+        /// <summary>
+        /// Error
+        /// </summary>
+        Error = 3,
+    }
+
+    /// <summary>
+    /// A class representing a message encountered while validating an OData Url.
+    /// </summary>
+    public class ODataUrlValidationMessage

```

Comment:
ODataUriError?

Reply:
It seems to defined a 'Severity',  So, "Message" enum member is confusing "Message" in  "ODataUrlValidationMessage"

Reply:
By design, since it could be an informational message, warning, or error.

---
In reply to: [434870792](https://github.com/OData/odata.net/pull/1769#discussion_r434870792) [](ancestors = 434870792)

Reply:
I used the same tags as Visual Studio - Error, Warning, Message.  The later is informational, one step below a warning.

---
In reply to: [434871493](https://github.com/OData/odata.net/pull/1769#discussion_r434871493) [](ancestors = 434871493)